src/libgap-api.c

```diff
@@ -422,6 +423,12 @@ int GAP_IsMatrixOrMatrixObj(Obj obj)
     return obj && DoFilter(IsMatrixOrMatrixObjFilt, obj) == True;
 }
 
+// Returns 1 if <obj> is a GAP matrix, 0 if not.
+int GAP_IsMatrix(Obj obj)
+{
+    return obj && DoFilter(IsMatrixFilt, obj) == True;
```

Comment:
`IsMatrix` is not a plain filter, but rather an and-filter, so calling `DoFilter` is wrong and the cause for the crash in the CI tests.
```suggestion
    return obj && CALL_1ARGS(IsMatrixFilt, obj) == True;
```
I suggest we change `GAP_IsMatrixOrMatrixObj` and `GAP_IsMatrixObj` similarly -- while `DoFilter` works for them right now, it's a bit risky, as any change to their definition would accidentally break this code.